### ISSUE TITLE
beforeSave has an inconsistent signature with save

### ISSUE THREAD

**Issue by ylonglew** [benhutchins/dyngoose#479]
https://github.com/benhutchins/dyngoose/blob/master/src/table.ts#L484
```
const allowSave = await this.beforeSave(conditions, meta)
```
https://github.com/benhutchins/dyngoose/blob/master/src/table.ts#L595

```
protected async beforeSave(meta?: any, conditions?: any): Promise<boolean> {
```

The order of `meta` and `conditions` should be corrected.

**Comment by benhutchins** [comment]
Good catch. Thank you for reporting this.

**Comment by benhutchins** [comment]
This was corrected as part of [2.16.5](https://github.com/benhutchins/dyngoose/releases/tag/v2.16.5)
